### PR TITLE
Update makefile to include building zip, checksum, and plan

### PR DESCRIPTION
**Pull request by mtps**
## Description

closes: #119

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit and integration [tests](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#testing)
  - [x] Not applicable - github workflow.
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
  - [x] Not applicable - no module changes.
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
  - [x] Not applicable - no go code changes.
- [x] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -36,11 +36,11 @@ Ref: https://keepachangelog.com/en/1.0.0/
 
 * Add support for creating, updating, removing, finding, and iterating over `Record`s #54
 * Add migration support for v039 account into v040 attributes module #100
-* Implemented v39 to v40 migration for name module.
```
Comment (mtps) [inline CHANGELOG.md:39]: @iramiller This was listed in the `v0.1.4` release but also duplicated in `Unreleased`. Does this have relevance to v0.1.5?

Comment (iramiller) [reply CHANGELOG.md:39]: I think that is a 0.1.4 thing only...

**Comment by iramiller** [comment]
Build linux fails ...
```
/bin/sh: 1: pushd: not found
```
:ugh:

**Comment by mtps** [comment]
> Build linux fails ...
> 
> ```
> /bin/sh: 1: pushd: not found
> ```
> 
> :ugh:

The tag was incorrectly used: should have been `v0.0.0` vs `refs/tags/v0.0.0`, I'm looking at it.

File: Makefile [hunk 2]
```diff
@@ -121,6 +121,50 @@ run: check-built run-config;
...
+.PHONY: build-release-bin
+build-release-bin: vendor build
+	mkdir -p $(RELEASE_BIN) && \
+	cp $(BUILDDIR)/provenanced $(RELEASE_BIN) && \
+	cp vendor/github.com/CosmWasm/wasmvm/api/libwasmvm.so $(RELEASE_BIN) && \
+	chmod +x $(RELEASE_BIN)/provenanced
+
+.PHONY: build-release-zip
+build-release-zip: build-release-bin
+	pushd $(BUILDDIR) && \
```
Comment (iramiller) [inline Makefile:162]: This makes it look like finishing #69 is going to be really easy ... just zip `./proto/provenance` and `./thirdparty/proto` into a proto-release-num.zip archive...

**Review by iramiller — APPROVED** [review]
Nice, LGTM.

**Comment by mtps** [comment]
Checksum added to plan on local make run, but not in gh actions. 👁️

**Review by dpederson-figure — DISMISSED** [review]

**Review by dpederson-figure — DISMISSED** [review]

**Comment by mtps** [comment]
Found the issue with shasum not getting into plan: the subshell created within the make command was located at the top level of the repo. This is NOT the case with local runs, with `act` runs, or with other subshell calls. Absolute pathing fixed the issue.

**Review by dpederson-figure — APPROVED** [review]

**Pull request merged**